hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/DefaultNearCacheManager.java

```diff
@@ -107,5 +132,96 @@ public void destroyAllNearCaches() {
             nearCacheMap.remove(nearCache.getName());
             nearCache.destroy();
         }
+        for (ScheduledFuture preloadTaskFuture : preloadTaskFutures) {
+            preloadTaskFuture.cancel(true);
+        }
+
+        if (storageTaskFuture != null) {
+            storageTaskFuture.cancel(true);
+        }
+    }
+
+    private void createAndSchedulePreloadTask(NearCache nearCache, DataStructureAdapter adapter) {
+        if (adapter != null) {
+            PreloadTask preloadTask = new PreloadTask(nearCache, adapter);
+            ScheduledFuture<?> scheduledFuture = executionService.schedule(preloadTask, 3, SECONDS);
+            preloadTask.setScheduledFuture(scheduledFuture);
+
+            preloadTaskFutures.add(scheduledFuture);
+        }
+    }
+
+    private void createAndScheduleStorageTask() {
+        if (storageTaskFuture == null) {
+            StorageTask storageTask = new StorageTask();
+            storageTaskFuture = executionService.scheduleWithRepetition(storageTask, 0, 1, SECONDS);
+        }
+    }
+
+    private class PreloadTask implements Runnable {
+
+        private final NearCache nearCache;
+        private final DataStructureAdapter adapter;
+
+        private volatile ScheduledFuture<?> scheduledFuture;
+
+        PreloadTask(NearCache nearCache, DataStructureAdapter adapter) {
+            this.nearCache = nearCache;
+            this.adapter = adapter;
+        }
+
+        @Override
+        public void run() {
+            nearCache.preload(adapter);
+            if (scheduledFuture != null) {
+                preloadTaskFutures.remove(scheduledFuture);
+            }
+        }
+
+        void setScheduledFuture(ScheduledFuture<?> scheduledFuture) {
+            this.scheduledFuture = scheduledFuture;
+        }
+    }
+
+    private class StorageTask implements Runnable {
+
+        private final long started = System.currentTimeMillis();
+        private final AtomicBoolean storageInProgress = new AtomicBoolean(false);
+
+        @Override
+        public void run() {
+            long now = System.currentTimeMillis();
+            for (NearCache nearCache : nearCacheMap.values()) {
+                if (!isScheduled(nearCache, now)) {
+                    continue;
+                }
+                if (storageInProgress.compareAndSet(false, true)) {
+                    try {
+                        nearCache.storeKeys();
+                    } finally {
+                        storageInProgress.set(false);
+                    }
+                }
+            }
+        }
+
+        private boolean isScheduled(NearCache nearCache, long now) {
+            NearCacheStats nearCacheStats = nearCache.getNearCacheStats();
+            NearCachePreloaderConfig preloaderConfig = nearCache.getPreloaderConfig();
+            if (nearCacheStats.getLastPersistenceTime() == 0) {
```

Comment:
All of the related stats are named `getPersistenceCount` or `getLastPersistence...`. And `write` could be a write to the Near Cache itself, not the disk. Maybe something with `store`, because we use that term for the `MapStore` (which is mostly a persistent layer as well).